src/mpid/ch4/shm/posix/posix_am.h

```diff
@@ -21,7 +21,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_am_enqueue_request(const void *am_hdr,
                                                             size_t iov_num_left, size_t data_sz,
                                                             MPIR_Request * sreq)
 {
-    MPIDI_POSIX_am_request_header_t *curr_sreq_hdr = NULL;
+    MPIDI_POSIX_am_request_header_t *curr_sreq_hdr = MPIDI_POSIX_AMREQUEST(sreq, req_hdr);
```

Comment:
Remind me why we need the `req_hdr` when it is not enqueued?

Reply:
It stores the pack buffer address, so that it can be correctly freed later.

Reply:
I see. From the code, either the original code was left incomplete, or we somehow broke the code at some point.

Reply:
Look that way.